src/app/pages/receive/receive-modal.tsx

```diff
@@ -0,0 +1,104 @@
+import toast from 'react-hot-toast';
+import { useLocation, useNavigate } from 'react-router-dom';
+
+import { Box, useClipboard } from '@stacks/ui';
+import { HomePageSelectors } from '@tests/selectors/home.selectors';
+import get from 'lodash.get';
+
+import { RouteUrls } from '@shared/route-urls';
+
+import { useAnalytics } from '@app/common/hooks/analytics/use-analytics';
+import { StxAvatar } from '@app/components/crypto-assets/stacks/components/stx-avatar';
+import { BaseDrawer } from '@app/components/drawer/base-drawer';
+import { BtcIcon } from '@app/components/icons/btc-icon';
+import { BtcStampsIcon } from '@app/components/icons/btc-stamps-icon';
+import { OrdinalIcon } from '@app/components/icons/ordinal-icon';
+import { useZeroIndexTaprootAddress } from '@app/query/bitcoin/ordinals/use-zero-index-taproot-address';
+import { useCurrentAccountNativeSegwitAddressIndexZero } from '@app/store/accounts/blockchain/bitcoin/native-segwit-account.hooks';
+import { useCurrentAccountStxAddressState } from '@app/store/accounts/blockchain/stacks/stacks-account.hooks';
+
+import { ReceiveItem } from './components/receive-item';
+import { ReceiveItems } from './components/receive-items';
+
+type ReceiveModal = 'full' | 'collectible';
+
+interface ReceiveModalProps {
+  type?: 'full' | 'collectible';
+}
+
+export function ReceiveModal({ type = 'full' }: ReceiveModalProps) {
+  const analytics = useAnalytics();
+  const location = useLocation();
+  const navigate = useNavigate();
+  const btcAddressNativeSegwit = useCurrentAccountNativeSegwitAddressIndexZero();
+  const stxAddress = useCurrentAccountStxAddressState();
+  const accountIndex = get(location.state, 'accountIndex', undefined);
+  const btcAddressTaproot = useZeroIndexTaprootAddress(accountIndex);
+
+  const { onCopy: onCopyBtc } = useClipboard(btcAddressNativeSegwit);
+  const { onCopy: onCopyStx } = useClipboard(stxAddress);
+  const { onCopy: onCopyOrdinal } = useClipboard(btcAddressTaproot);
+
+  function copyToClipboard(copyHandler: () => void, tracker = 'copy_address_to_clipboard') {
+    void analytics.track(tracker);
+    toast.success('Copied to clipboard!');
+    copyHandler();
+  }
+
+  const title = type === 'full' ? 'Select asset to receive' : 'Add collectible';
+
+  return (
+    <BaseDrawer title={title} isShowing onClose={() => navigate('../')}>
+      <Box mx="extra-loose">
+        {type === 'full' && (
+          <ReceiveItems title="Tokens">
+            <ReceiveItem
```

Comment:
`ReceiveItem` is now more generic and we can use the same code for the standard `Receive` modal and what was `ReceiveCollectibleModal` by passing in a type: `<ReceiveModal type="collectible" />`